core/src/main/java/ar/com/wolox/wolmo/core/permission/PermissionManager.kt

```diff
@@ -0,0 +1,121 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package ar.com.wolox.wolmo.core.permission
+
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Handler
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import androidx.core.content.ContextCompat
+import android.util.SparseArray
+
+import ar.com.wolox.wolmo.core.di.scopes.ApplicationScope
+
+import java.util.ArrayList
+
+import javax.inject.Inject
+
+/** Helper class to handle Android's runtime permissions */
+@ApplicationScope
+class PermissionManager @Inject
+constructor(private val context: Context, private val requestListeners: SparseArray<PermissionListener>) {
+
+    private var requestCount = 1
+
+    /**
+     * Request one or more Android's runtime permissions using [fragment] to perform the
+     * request, [listener] to receive the response of the request (can be null if not
+     * interested in the response) and [permissions] ( One or more [String] objects with
+     * the permissions to be requested. Returns **true** if every requested permission
+     * was already granted, **false** otherwise.
+     */
+    fun requestPermission(fragment: Fragment, listener: PermissionListener?, vararg permissions: String): Boolean {
+        val ungrantedPermissions = filterUngranted(*permissions)
+        if (ungrantedPermissions.isNotEmpty()) {
+            fragment.requestPermissions(ungrantedPermissions, requestCount)
+            requestListeners.put(requestCount++, listener)
+            return false
+        }
+        listener?.onPermissionsGranted()
+        return true
+    }
+
+    /**
+     * Request one or more Android's runtime permissions using an [activity] to perform the
+     * request, [listener] to receive the response of the request (can be null if not
+     * interested in the response) and [permissions] ( One or more [String] objects with
+     * the permissions to be requested. Returns **true** if every requested permission
+     * was already granted, **false** otherwise.
+     */
+    fun requestPermission(activity: Activity, listener: PermissionListener?, vararg permissions: String): Boolean {
+        val ungrantedPermissions = filterUngranted(*permissions)
+        if (ungrantedPermissions.isNotEmpty()) {
+            ActivityCompat.requestPermissions(activity, ungrantedPermissions, requestCount)
```

Comment:
The only line changed from the previous method is this. We should check how to save this code repetition.